locallib.php

```diff
@@ -346,24 +359,112 @@ function zoom_get_sessions_for_display($meetingid) {
     return $sessions;
 }
 
+/**
+ * Get the next occurrence of a meeting.
+ *
+ * @param stdClass $zoom
+ * @return int The timestamp of the next occurrence of a recurring meeting or
+ *             0 if this is a recurring meeting without fixed time or
+ *             the timestamp of the meeting start date if this isn't a recurring meeting.
+ */
+function zoom_get_next_occurrence($zoom) {
+    global $DB;
+
+    // Prepare an ad-hoc request cache as this function could be called multiple times throughout a request
+    // and we want to avoid to make duplicate DB calls.
+    $cacheoptions = array(
+            'simplekeys' => true,
+            'simpledata' => true
+    );
+    $cache = cache::make_from_params(cache_store::MODE_REQUEST, 'zoom', 'nextoccurrence', array(), $cacheoptions);
+
+    // If the next occurrence wasn't already cached, fill the cache.
+    if (($cachednextoccurrence = $cache->get($zoom->id)) === false) {
+        // If this isn't a recurring meeting.
+        if (!$zoom->recurring) {
+            // Use the meeting start time.
+            $cachednextoccurrence = $zoom->start_time;
+
+            // Or if this is a recurring meeting without fixed time.
+        } else if ($zoom->recurrence_type == ZOOM_RECURRINGTYPE_NOTIME) {
+            // Use 0 as there isn't anything better to return.
+            $cachednextoccurrence = 0;
+
+            // Otherwise we have a recurring meeting with a recurrence schedule.
+        } else {
+            // Get the calendar event of the next occurrence.
+            $selectclause = "modulename = :modulename AND instance = :instance AND (timestart + timeduration) >= :now";
+            $selectparams = array('modulename' => 'zoom', 'instance' => $zoom->id, 'now' => time());
+            $nextoccurrence = $DB->get_records_select('event', $selectclause, $selectparams, 'timestart ASC', 'timestart', 0, 1);
+
+            // If we haven't got a single event.
+            if (count($nextoccurrence) != 1) {
```

Comment:
I do not like using count() if we don't care about the actual count. I have two different ideas.

Option 1: If `$nextoccurrence` is an array, then we can use `empty()`
```suggestion
            if (empty($nextoccurrence)) {
```

Option 2: If `empty()` returns false because `$nextoccurrence` is an object that confuses `empty()`
```suggestion
            if (count($nextoccurrence) === 0) {
```

After all, if there's one event or two events, we're still only going to pick the first one. And in theory, the limit arguments passed to get_records_select will help ensure sure there's only the first one.

Reply:
updated to use `empty()` here